go/vt/vtgate/engine/fk_cascade.go

```diff
@@ -185,7 +188,19 @@ func (fkc *FkCascade) executeNonLiteralExprFkChild(ctx context.Context, vcursor
 
 		// Next, we need to copy the updated expressions value into the bind variables map.
 		for _, info := range child.NonLiteralInfo {
-			bindVars[info.UpdateExprBvName] = sqltypes.ValueBindVariable(row[info.UpdateExprCol])
+			// Type case the value to that of the column that we are updating.
+			// This is required for example when we receive an updated float value of -0, but
+			// the column being updated is a varchar column, then if we don't coerce the value of -0 to
+			// varchar, MySQL ends up setting it to '0' instead of '-0'.
+			finalVal := row[info.UpdateExprCol]
+			var err error
+			if !finalVal.IsNull() {
+				finalVal, err = evalengine.CoerceTo(finalVal, selectionRes.Fields[info.ExprCol].Type)
+			}
+			if err != nil {
+				return err
+			}
```

Comment:
nit: error check can happen inside the `if` condition.

Reply:
Fixed